src/encoding.rs

```diff
@@ -27,7 +27,7 @@ pub fn encode_varint<B>(mut value: u64, buf: &mut B)
 where
     B: BufMut,
 {
-    loop {
+    for _ in 0..10 {
```

Comment:
Can you explain the constant 10? Maybe a static assert, formula or documentation?

Reply:
is it not sufficiently covered by the comment already on this function? i can add it, just wondering

Reply:
Yeah, you are probably right. I saw that comment and started looking into why 10 bytes is correct. I took me a moment to understand.

Reply:
protobuf's varints are fairly thoroughly documented to be 10 bytes maximum, yes. i mean they could be 9 but they decided not to do that. i believe every minimal length protobuf varint that is 10 bytes always has 0x01 as its last byte value, which is rather a waste considering how many situations cause them to become maximum length this way. the burden of legacy i guess